### PR TITLE
Port to protocol v3

### LINKED ISSUE #78: Add example documentation for subscribing to a sensor
**Is your feature request related to a problem? Please describe.**
I was struggling to figure out how to subscribe to sensors on the drone. (If it was polling at a fixed rate or if you get connected to a stream). After finding the example of print status `examples/print_status.py` things got easier. 

**Describe the solution you'd like**
Add the drone state example to the documentation to show how you can subscribe to a sensor on the drone.

**Additional context**
image

image

### PR DESCRIPTION
**Pull request by sindrehan**
Todo:
- [x] Test client_info generation on Windows/OSX
- [x] Write instructions for upgrading from v1.0 to v2.0 (due to breaking changes)
- [ ] Fix slow unit tests (likely due to mocking)
- [x] Add depth sensor example to docs (To fix #78)
- [x] Remove slave-mode functionality in favor of the new functions for taking/releasing control of the drone
- [x] Update camera overlay functions
- [x] Update camera parameter functions
- [x] Add property for still picture bitrate
- [x] Update motion control functions
- [x] ~Figure out a good interface for dealing with guest port peripherals~ #129, #130, #131, #132, #133
- [x] Debug thread not closing properly
- [x] Fix integration tests
- [x] ~Add new overlay parameters (thickness gauge, etc)~ #127 
- [x] ~Update documentation for new log file format~ #128
- [x] Verify Blunux > 3.1 on connection
- [x] Find a better way to deal with trying to read a telemetry message that hasn't been received yet (Currently just raises a `KeyError`)
- [x] Add function for controlling topic publish rate
- [x] Deserialize telemetry messages when getting them

Breaking changes:
  - Removed slave-mode
  - Update range for lights from 0-255 to 0-1
  - Require Blunux v3.1 or newer
  - Error flags is a dict of bools instead of int
  - active streams is a dict with main and guestport
  - Water density is kg/L instead of g/L
  - Camera stabilization is enable/disable instead of toggle
  - Renamed tilt_speed to tilt_velocity
  - State of charge range is 0..1 instead of 0..100
  - Custom overlay classes have been replaced with enums from blueye.protocol
  - Properties that read telemetry data (lights, tilt_angle, dept, pose, battery_state_of_charge, error_flags, active_video_streams, auto_depth_active, auto_heading_active) will now return `None` if no telemetry message has been received yet from the drone.

### CODE REVIEW

File: blueye/sdk/constants.py [hunk 1]
```diff
@@ -5,9 +5,9 @@
...
 class WaterDensities:
     """
-    Various typical densities for salt water (in grams/liter)
+    Various typical densities for salt water (in kilograms/liter)
     """
 
-    fresh = 997
-    brackish = 1011
-    salty = 1025
+    fresh = 0.997
```
Comment (johannesschrimpf) [inline blueye/sdk/constants.py:11]: Should we rather use enum here:
https://docs.python.org/3/library/enum.html

File: blueye/sdk/connection.py [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+
+    def take_still_picture(self):
+        msg = blueye.protocol.TakePictureCtrl()
+        self.messages_to_send.put(msg)
+
+
+class ReqRepClient(threading.Thread):
+    def __init__(self, parent_drone: "blueye.sdk.Drone", context: zmq.Context = None):
+        super().__init__(daemon=True)
+        self.context = context or zmq.Context().instance()
```
Comment (johannesschrimpf) [inline blueye/sdk/connection.py:159]: I guess context, socket, port, requests_to_send should start with underscore here as they are private variables.

File: blueye/sdk/connection.py [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+    def stop(self):
+        """Stop the watchdog thread started by run()"""
+        self._exit_flag.set()
+
+
+class TelemetryClient(threading.Thread):
+    def __init__(self, parent_drone: "blueye.sdk.Drone", context: zmq.Context = None):
+        super().__init__(daemon=True)
+        self._parent_drone = parent_drone
+        self.context = context or zmq.Context().instance()
```
Comment (johannesschrimpf) [inline blueye/sdk/connection.py:50]: I think we can make many of the variables here private.

File: blueye/sdk/constants.py (hunk 1 above)
Comment (sindrehan) [reply blueye/sdk/constants.py:11]: I don't know if an enum makes more sense, the `water_density` function supports setting it to any other positive float, those three values are just the default ones we use in the app.

Comment (johannesschrimpf) [reply blueye/sdk/constants.py:11]: It seem you can define float enums:
https://stackoverflow.com/questions/69716107/can-i-use-an-enum-with-floating-point-values-and-comparators-in-python-3-9-and-s

**Pull request merged**